optimum/onnxruntime/__init__.py

```diff
@@ -11,56 +11,67 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
-from enum import Enum
+from typing import TYPE_CHECKING
 
+from transformers.utils import _LazyModule
 
-class ORTQuantizableOperator(Enum):
-    # Common ops
-    Gather = "Gather"
-    Transpose = "Transpose"
-    EmbedLayerNormalizationQuant = "EmbedLayerNormalization"
 
-    # QLinearOps
-    Conv = "Conv"
-    MatMul = "MatMul"
-    Add = "Add"
-    Mul = "Mul"
-    Relu = "Relu"
-    Clip = "Clip"
-    LeakyRelu = "LeakyRelu"
-    Sigmoid = "Sigmoid"
-    MaxPool = "MaxPool"
-    GlobalAveragePool = "GlobalAveragePool"
-    Split = "Split"
-    Pad = "Pad"
-    Reshape = "Reshape"
-    Squeeze = "Squeeze"
-    Unsqueeze = "Unsqueeze"
-    Resize = "Resize"
-    AveragePool = "AveragePool"
-    Concat = "Concat"
+_import_structure = {
+    "configuration": ["ORTConfig"],
+    "model": ["ORTModel"],
+    "modeling_ort": [
+        "ORTModelForCausalLM",
+        "ORTModelForFeatureExtraction",
+        "ORTModelForImageClassification",
+        "ORTModelForQuestionAnswering",
+        "ORTModelForSequenceClassification",
+        "ORTModelForTokenClassification",
+    ],
+    "modeling_seq2seq": ["ORTModelForSeq2SeqLM"],
+    "optimization": ["ORTOptimizer"],
+    "quantizable_operators": [
+        "ORTQuantizableOperator",
+        "AUTO_MINIMUM_SUPPORTED_ONNX_OPSET",
+        "ORT_DEFAULT_CHANNEL_FOR_OPERATORS",
+        "ORT_FULLY_CONNECTED_OPERATORS",
+    ],
+    "quantization": ["ORTQuantizer"],
+    "trainer": ["ORTTrainer"],
+    "trainer_seq2seq": ["ORTSeq2SeqTrainer"],
+    "utils": ["ONNX_DECODER_NAME", "ONNX_DECODER_WITH_PAST_NAME", "ONNX_ENCODER_NAME", "ONNX_WEIGHTS_NAME"],
+}
 
 
-AUTO_MINIMUM_SUPPORTED_ONNX_OPSET = None
+# Direct imports for type-checking
+if TYPE_CHECKING:
```

Comment:
Just out of curiosity: what does this direct import enable?

Reply:
It is only for type checking, otherwise modules are imported using `_LazyModule`. Not sure that answers your question though :thinking: